featuretools/feature_base/feature_base.py

```diff
@@ -410,7 +413,7 @@ def from_dictionary(cls, arguments, entityset, dependencies, primitives_deserial
         relationship_path = [Relationship.from_dictionary(r, entityset)
                              for r in arguments['relationship_path']]
         child_entity = relationship_path[0].child_entity
-        return cls(base_feature, child_entity, relationship_path=relationship_path)
+        return cls(base_feature, child_entity, relationship_path=relationship_path, name=arguments['name'])
```

Comment:
Let's be explicit with the other 2 arguments as well, `base_feature, child_entity`